video/out/vulkan/context_display.c

```diff
@@ -0,0 +1,423 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "options/m_config.h"
+
+#include "context.h"
+#include "utils.h"
+
+struct vulkan_display_opts {
+    char *display_spec;
+};
+
+static bool parse_display_spec(struct mp_log *log,
```

Comment:
This function seems weirdly written for a number of reasons. It seems to contain logic for dealing with partial strings, but a string like "0" would just get rejected by the function unless it contains a `:` as well.

I recommend finding another way to write this in a way that's a less confusing mess of conditionals, and does the right thing on partial strings. Whatever that "right thing is", I'm not sure what the intended logic is, but on my system at least, specifying the mode has the highest priority, followed by the display/plane, because there's only one of the latter. So I'd expect this being in the order `[[display:]plane:]mode`, with just e.g. `5` being interpreted as `0:0:5`. :woman_shrugging:

Is this a common form of identifier or something you made up?